pandas/io/json/_json.py

```diff
@@ -101,20 +91,23 @@ def to_json(
     if lines:
         s = convert_to_line_delimits(s)
 
-    if isinstance(path_or_buf, str):
-        fh, handles = get_handle(path_or_buf, "w", compression=compression)
+    if path_or_buf is not None:
+        ioargs = get_filepath_or_buffer(
+            path_or_buf,
+            compression=compression,
+            mode="wt",
+            storage_options=storage_options,
+        )
+        handle_args = get_handle(
```

Comment:
combining `get_filepath_or_buffer` (mostly used to open URLs) and `get_handle` (compression, opening files, wrapping bytes, and memory mapping) in some way would make a lot of sense. I feel that calling `get_filepath_or_buffer` inside `get_handle` would be a good solution. But I would need to first re-visit all places that call `get_filepath_or_buffer` and `get_handle` to make sure that this satisfies everyone.

@jreback Do you prefer to have this in this PR or in a followup? If adding more changes to this PR is okay from a review perspective, I'm tempted do add it to this PR (instead of touching `IOArgs` twice). I'll probably have time to combine them by the end of the next weekend. Is there a deadline/feature freeze for 1.2 upcoming?

Reply:
i think a follow on or would be easier to review 

1.2 is schedule for end of nov so have a little time